### ISSUE TITLE
Crash on 1.20.1 on Quilt

### ISSUE THREAD

**Issue by NickAcPT** [BasiqueEvangelist/DynReg#3]
While trying to get DynReg working to experiment with it, my game just crashes when joining a world. [Game Log](https://mclo.gs/DFu9khn).

I can only seem to reproduce the crash with Quilted Fabric API. When using normal Fabric API under Quilt doesn't crash on join.

**Comment by BasiqueEvangelist** [comment]
Yeah, I didn't really test the Quilt support on 1.20 yet :sweat_smile: 
Will try to fix™

**Comment by BasiqueEvangelist** [comment]
Can't reproduce in dev env

**Comment by NickAcPT** [comment]
> Can't reproduce in dev env

@BasiqueEvangelist Yeah, I figured. I've tried reproducing it in the dev environment too without much success either.

However, here's [a PrismLauncher instance](https://github.com/BasiqueEvangelist/DynReg/files/11893216/dynreg-repro.zip) I exported to try and reproduce it.

The issue occurs as soon as I press the Singleplayer button and go to create a world for the first time. No resourcepacks or anything included. [Here](https://mclo.gs/sRAUHPd) is a log with `-Dmixin.debug=true` to help diagnose the issue.

If needed, feel free to direct message me on Discord and I'll try helping with reproducing the issue.

EDIT: [Here](https://mclo.gs/DFu9khn)'s a log without mixin.debug. Weirdly enough, when the debug is enabled, it's complaining about the Fabric hook not being applied.. which is stupid since it's not the hook you're trying to apply.. huh
